tools/analyzer_plugin/test/integration/assists/toggle_statefulness_test.dart

```diff
@@ -0,0 +1,151 @@
+import 'dart:async';
+
+import 'package:analyzer_plugin/utilities/assist/assist.dart';
+import 'package:over_react_analyzer_plugin/src/assist/toggle_stateful.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/assist_test_base.dart';
+import 'boilerplate_assist_utils.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AddStatefulnessAssist);
+    defineReflectiveTests(RemoveStatefulnessAssist);
+  });
+}
+
+@reflectiveTest
+class AddStatefulnessAssist extends AssistTestBase with BoilerplateAssistTestStrings {
+  static int expectedPriority = ToggleComponentStatefulness.statefulAssistPriority;
+
+  @override
+  AssistKind get assistKindUnderTest => ToggleComponentStatefulness.makeStateful;
+
+  Future<void> test_noAssistOnFactory() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(BoilerplateAssistTestStrings.fileName, generatedSource);
+    var selection = createSelection(source,
+        generatedSource.replaceFirst(r'UiFactory<FooProps> Foo = _$Foo;', r'UiFactory<FooProps> #Foo# = _$Foo;'));
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnProps() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(BoilerplateAssistTestStrings.fileName, generatedSource);
+    var selection = createSelection(
+        source, generatedSource.replaceFirst('mixin FooProps on UiProps {}', 'mixin #FooProps# on UiProps {}'));
+    await expectNoAssist(selection);
+  }
+
+  Future<void> test_noAssistOnOldBoilerplate() async {
+    String oldBoilerplate({bool withSelection = false}) => '''
+import 'package:over_react/over_react.dart';
+part 'test.over_react.g.dart';
+
+@Factory()
+UiFactory<TestProps> Test = _\$Test;
+
+@Props()
+class _\$TestProps extends UiProps {}
+
+@Component()
+class ${withSelection ? '#TestComponent#' : 'TestComponent'} extends UiComponent<TestProps> {
+  @override
+  render() {}
+}
+''';
+
+    var source = newSource(BoilerplateAssistTestStrings.fileName, oldBoilerplate());
+    var selection = createSelection(source, oldBoilerplate(withSelection: true));
+    await expectNoAssist(selection);
+  }
```

Comment:
Can we add a couple test cases for edge cases?

- non-supported component base class
- running this assist in a file with no component

Reply:
For sure! How would the non-supported base class differ from https://github.com/Workiva/over_react/pull/580/files#diff-a479f45e42c16ab64749a5016a13af57R43? Is there another base class we should test against?

But added a test with a file with no component!

Reply:
My thought process was to test `_getNewBase ` and the related code path: https://github.com/Workiva/over_react/blob/master/tools/analyzer_plugin/lib/src/assist/toggle_stateful.dart#L49-L52

since for old boilerplate, the assist will short-circuit by returning false in `initializeAssistApi`, since `ComponentDeclarationAssistApi` doesn't support legacy boilerplate

Reply:
Ohh, duhh, haha. Good call, adding that in the other branch!